mqttwarn/services/azure-iot.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from six import string_types
+
+__author__    = 'Morten Høybye Frederiksen <morten()mfd-consult.dk>'
+__copyright__ = 'Copyright 2016-2020 Morten Høybye Frederiksen'
+__license__   = 'Eclipse Public License - v 1.0 (http://www.eclipse.org/legal/epl-v10.html)'
+
+from builtins import str
+import paho.mqtt.publish as mqtt
+import paho.mqtt.client as mqttclient
+import ssl
+
+def plugin(srv, item):
+    srv.logging.debug("*** MODULE=%s: service=%s, target=%s", __file__, item.service, item.target)
+
+    # addrs is a list[] containing device id and sas token
+    deviceid, sastoken = item.addrs
+
+    # iot hub name is stored in config
+    iothubname = item.config['iothubname']
+
+    # connect...
+    params = {
+        'hostname': iothubname + ".azure-devices.net",
+        'port': 8883,
+        'protocol': mqttclient.MQTTv311,
+        'qos': 0,
```

Comment:
Is it possible to use an elevated QoS value here? If so, we might want to make configurable:
```
'qos': int(item.config.get('qos', 0))
```

Reply:
Sure. QoS 2 is not allowed, however, so I've added a check.